unit/java_bytecode/java_object_factory/gen_nondet_string_init.cpp

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************\
+
+ Module: Java string library preprocess.
+         Test for converting an expression to a string expression.
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <util/expr.h>
+#include <util/std_code.h>
+#include <util/namespace.h>
+#include <java_bytecode/java_object_factory.h>
+#include <java_bytecode/java_bytecode_language.h>
+#include <java_bytecode/java_root_class.h>
+#include <langapi/language_util.h>
+#include <langapi/mode.h>
+#include <iostream>
+
+TEST_CASE("Generate string object")
+{
+  source_locationt loc;
+  symbol_tablet symbol_table;
+  register_language(new_java_bytecode_language);
+
+  // Add java.lang.Object to symbol table
+  symbolt jlo_sym;
+  jlo_sym.name="java::java.lang.Object";
+  jlo_sym.type=struct_typet();
+  jlo_sym.is_type=true;
+  java_root_class(jlo_sym);
+  bool failed=symbol_table.add(jlo_sym);
+  CHECK_RETURN(!failed);
+
+
+  // Add java.lang.Object
+  symbolt new_symbol;
+  new_symbol.base_name="java::java.lang.Object";
+  new_symbol.pretty_name="java::java.lang.Object";
+  new_symbol.name="java::java.lang.Object";
+  new_symbol.type=struct_typet();
+  new_symbol.mode=ID_java;
+  new_symbol.is_type=true;
+
+  symbolt *class_symbol;
+  symbol_table.move(new_symbol, class_symbol);
+
+
+  // Add java.lang.String to symbol table
+  java_string_library_preprocesst preprocess;
+  preprocess.add_string_type("java.lang.String", symbol_table);
+  namespacet ns(symbol_table);
+
+  // Declare a String named arg
+  symbol_typet java_string_type("java::java.lang.String");
+  symbol_exprt expr("arg", java_string_type);
+  // Generate initialisation code for arg in code
+  const codet code=
```

Comment:
To be clear, this is essentially the function under test. Consider using `GIVEN`/`THEN` constructs to help separate set up code with checking code (particular as both the setup and checking code are non-trivial)